src/infra/dex/balancer/dto.rs

```diff
@@ -189,6 +189,7 @@ impl Chain {
             eth::ChainId::Mainnet => Ok(Self::Mainnet),
             eth::ChainId::Gnosis => Ok(Self::Gnosis),
             eth::ChainId::ArbitrumOne => Ok(Self::Arbitrum),
+            eth::ChainId::Base => Ok(Self::Base),
             unsupported => Err(Error::UnsupportedChainId(unsupported)),
```

Comment:
Instead of using `unsupported` as the catch-all term we should probably explicitly list the unsupported networks. That way the compiler would have complained about the match statement here when you added the `Base` variant.

Reply:
The actual comment was not addressed, though.
My point was to change the match statement to ensure that the compiler yells at us whenever we add a new chain. So in our case:
```rust
eth::ChainId::Goerli => Err(Error::UnsupportedChain(chain_id)),
```